setuptools/dist.py

```diff
@@ -57,6 +57,8 @@ def write_pkg_file(self, file):
     file.write('License: %s\n' % self.get_license())
     if self.download_url:
         file.write('Download-URL: %s\n' % self.download_url)
+    for project_url in self.project_urls.items():
```

Comment:
Should this be changed to operate on a sorted list of keys instead so that output order remains consistent between Python versions?

Reply:
Presumably the order here is going to be the same as it is displayed in on PyPi?

The user will probably want to present a certain order that is not alphabetical. If the config is in code (i.e. setup.py) the user will have to make sure this preserves ordering (i.e. use a collections.OrderedDict, or dict on Python3.6+).

However, in the setup.cfg the dict creation is done by setuptools, and it isn't ordered. As python2.6 is not supported anymore, we can replace [the dictionary creation](https://github.com/pypa/setuptools/pull/1210/files#diff-360050608fe825a457c8093298edf207R219) with a collections.OrderedDict to preserve the order.

Reply:
In light of #458, I suspect it's going to be a hard requirement that any package metadata be captured or stored in a deterministic order. Even in Python 3.6, the ordering "[should not be relied upon](https://docs.python.org/3.6/whatsnew/3.6.html#new-dict-implementation)." Therefore, I advise that the user be required to provide an ordering or that the metadata always sort the keys before storing the values.

Reply:
Okay, so dict type structures from setup.cfg will need switching to collections.ordereddict under the covers anyway, and I shouldn't need to make any corresponding adjustments to this PR to accomodate?

As for "same as it is displayed in on PyPI" that's a bit uncertain at the moment. I agree that PyPI should make some attempt to preserve ordering of this, but my test upload is showing the reverse order from what's in the sdist PKG-INFO and wheel METADATA files on the cheeseshop (I haven't been able to sufficiently untangle the source there but I have a feeling order is being determined by DB query results). By contrast, the implementation in warehouse exhibits the same sort of list-of-tupes-fed-into-unordered-dict design as here (and is also not working at the moment, per discussion in https://github.com/pypa/warehouse/issues/1419 ).